hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SStream.scala

```diff
@@ -1,15 +1,47 @@
 package is.hail.types.physical.stypes.interfaces
 
 import is.hail.annotations.Region
-import is.hail.asm4s.{Settable, TypeInfo, Value}
-import is.hail.expr.ir.EmitCodeBuilder
+import is.hail.asm4s._
+import is.hail.expr.ir.{EmitCode, EmitCodeBuilder, EmitMethodBuilder, IEmitCode}
 import is.hail.expr.ir.streams.StreamProducer
 import is.hail.types.physical.PType
 import is.hail.types.physical.stypes._
 import is.hail.types.virtual.{TStream, Type}
 import is.hail.types.{RIterable, TypeWithRequiredness}
+import is.hail.utils.FastIndexedSeq
+
+trait NoBoxLongIterator {
+  def init(partitionRegion: Region, elementRegion: Region)
+
+  def next(): Long // -1L if EOS, 0 if missing
```

Comment:
-1 is a valid address (interpreted as unsigned), as discussed

Reply:
oops, didn't address this yet

Reply:
wait yes I did, just needed to fix up a comment